edge/edge.go

```diff
@@ -60,26 +66,44 @@ func (manager *Manager) Start() error {
 		EdgeID:                  manager.agentOptions.EdgeID,
 		PollFrequency:           agent.DefaultEdgePollInterval,
 		InactivityTimeout:       manager.agentOptions.EdgeInactivityTimeout,
-		InsecurePoll:            manager.agentOptions.EdgeInsecurePoll,
 		PortainerURL:            manager.key.PortainerInstanceURL,
 		EndpointID:              manager.key.EndpointID,
 		TunnelServerAddr:        manager.key.TunnelServerAddr,
 		TunnelServerFingerprint: manager.key.TunnelServerFingerprint,
 		ContainerPlatform:       manager.containerPlatform,
 	}
 
-	log.Printf("[DEBUG] [internal,edge] [api_addr: %s] [edge_id: %s] [poll_frequency: %s] [inactivity_timeout: %s] [insecure_poll: %t]", pollServiceConfig.APIServerAddr, pollServiceConfig.EdgeID, pollServiceConfig.PollFrequency, pollServiceConfig.InactivityTimeout, pollServiceConfig.InsecurePoll)
+	log.Printf("[DEBUG] [internal,edge] [api_addr: %s] [edge_id: %s] [poll_frequency: %s] [inactivity_timeout: %s] [insecure_poll: %t]", pollServiceConfig.APIServerAddr, pollServiceConfig.EdgeID, pollServiceConfig.PollFrequency, pollServiceConfig.InactivityTimeout, manager.agentOptions.EdgeInsecurePoll)
 
-	stackManager, err := newStackManager(manager.key.PortainerInstanceURL, manager.key.EndpointID, manager.agentOptions.EdgeID, pollServiceConfig.InsecurePoll)
+	stackManager, err := stack.NewStackManager(
+		client.NewPortainerClient(
+			manager.key.PortainerInstanceURL,
+			manager.key.EndpointID,
+			manager.agentOptions.EdgeID,
+			GetNewHttpClient(10, manager.agentOptions),
```

Comment:
maybe it's cleanr to pass the `manager.agentOptions` as param and call `GetNewHttpClient` internally

Reply:
atm, there are too many things created internally, breaking both the sharing, and hiding the high number of internal members - so I've been pulling it appart and making it more obvious.
Annoyingly, what I was originally trying to acheive started to be too big a refactor, so I'm currently doing the async thing in at a PortainerClient level, so its possibly true.
Main thrust is that the actual http requests are now in the pluggable PortainerClient, and this set of 3 managers all use the same code - except for the different timeouts, they would use the same PortainerClient instance - but that's something where I don't have time to work out the consequenses of the two 10second constants vs the User tuned one.

Reply:
yep I get that, what I had in mind was to at least pass the `timeout` value as a parameter and use that to call `GetNewHttpClient` internally, but it surely can be changed later, it's mostly a readability issue